Update documentation with groups information (#744)
TulipaEnergy/TulipaEnergyModel.jl: docs/src/how-to-use.md

## [Defining a group of assets](@id group-setup)

A group of assets refers to a set of assets that share certain constraints. For example, the investments of a group of assets may be capped at a maximum value, which represents the potential of a specific area that is restricted in terms of the maximum allowable MW due to limitations on building licenses.

In order to define the groups in the model, the following steps are necessary:

1. Create a group in the [`groups-data.csv`](@ref schemas) file by defining the `name` property and its parameters.

> 2. In the file [`assets-data.csv`](@ref schemas), assign to the assets that belong to the group the `name` of the group they belong in the `group` parameter/column.

Reviewer: ```suggestion
2. In the file [`assets-data.csv`](@ref schemas), assign assets to the group by putting the `name` in the `group` parameter/column.
```

    > **Note:**
    > A missing value in the parameter `group` in the [`assets-data.csv`](@ref schemas) means that the asset does not belong to any group.

> Groups are useful to represent several common constraints, the following group constraints are the ones available in the model.

Reviewer: ```suggestion
Groups are useful to represent several common constraints, the following group constraints are available.
```

### [Setting up a maximum or minimum investment limit for a group](@id investment-group-setup)

The mathematical formulation of the maximum and minimum investment limit for group constraints is available [here](@ref investment-group-constraints). The parameters to set up these constraints in the model are in the [`groups-data.csv`](@ref schemas) file.

> -   `invest_method = true`. This parameter enables the model to use the investment group constraints.

Reviewer: Can we change it to "invest_limit" or "limit_investment" instead of method?
Method to me means the way of investing or something.
Or "constrain_investment" maybe?

Author: mmm...the method is Juha's style that we are trying to mimic...I don't have a strong opinion on that...Maybe we can ask @g-moralesespana what he prefers as name for the parameter...

Author: @clizbe I have created issue #747 to add the possibility of an asset belonging to more than one group and review the naming of the constraints' activation. So, we can leave it like this now and discuss it later when tackling the issue. Is that ok?

Let's now explore which assets are in each group. To do so, we can take a look at the assets-data.csv file:

```@example display-group-setup
input_asset_file = "../../test/inputs/Norse/assets-data.csv" # hide
assets = CSV.read(input_asset_file, DataFrame, header = 2) # hide
assets = assets[.!ismissing.(assets.group), [:name, :type, :group, :investable]] # hide
```

Here we can see that the assets `Asgard_Solar` and `Midgard_Wind` belong to the `renewables` group, while the assets `Asgard_CCGT` and `Midgard_CCGT` belong to the `ccgt` group.

> **Note:**

> > Remember, if the `invest_method` is set to `true` for the groups, then the assets must also have the `investable` parameter set to `true`. This is necessary for formulating the group constraints correctly. If the assets in the group are not investable but the group has a minimum limit to fulfill, it can lead to an infeasible model. For example, if there are no investable assets in the group and the group has a minimum investment constraint, it would be impossible to meet the constraint without any available investments.

Reviewer: ```suggestion
> If the group has a `min_investment_limit`, then assets in the group have to allow investment (`investable = true`) for the model to be feasible. If the assets are not `investable` then they cannot satisfy the minimum constraint.
```
